actionpack/lib/action_dispatch/middleware/server_timing.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "active_support/notifications"
+
+module ActionDispatch
+  class ServerTiming
+    SERVER_TIMING_HEADER = "Server-Timing"
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      events = []
+      ActiveSupport::Notifications.subscribe(/.*/) do |*args|
+        events << ActiveSupport::Notifications::Event.new(*args)
+      end
```

Comment:
Need to unsubscribe as well?

Reply:
You're right. Just added that 9d4d4558dfa4d4eb35d0ddff5b597cdccb1019c9.